pkg/controller/statusmanager/pod_status.go

```diff
@@ -98,77 +99,96 @@ func SetNodeCondition(conditions *[]corev1.NodeCondition, newCondition corev1.No
-		sharedInfo.LastNetworkAvailable[nodeName] = *startedAt
+	existingCondition := FindNodeStatusCondition(node.Status.Conditions, "NetworkUnavailable")
+	if existingCondition == nil {
+		log.Info(fmt.Sprintf("Expecting node %s ready=%v, observed nil", nodeName, desireReady))
+	} else if desireReady && existingCondition.Status != corev1.ConditionFalse {
+		log.Info(fmt.Sprintf("Expecting node %s ready=true, observed false", nodeName))
+	} else if !desireReady && existingCondition.Status == corev1.ConditionFalse {
+		log.Info(fmt.Sprintf("Expecting node %s ready=false, observed true", nodeName))
 	} else {
-		delete(sharedInfo.LastNetworkAvailable, nodeName)
+		return reconcile.Result{}, nil
 	}
-	_, changed := status.combineNode(nodeName, ready, reason, message)
-	if !changed {
-		log.Info("Node condition is not changed")
-		return
-	}
-	log.Info(fmt.Sprintf("Setting status NetworkUnavailable to %v for node %s", !ready, nodeName))
+	return reconcile.Result{RequeueAfter: operatortypes.DefaultResyncPeriod}, nil
+}
+
+func (status *StatusManager) setNodeNetworkUnavailable(nodeName string, ready bool, startedAt *time.Time, reason string, message string, sharedInfo *sharedinfo.SharedInfo) (reconcile.Result, error) {
+	log.V(1).Info(fmt.Sprintf("Setting status NetworkUnavailable to %v for node %s", !ready, nodeName))
 	if !ready {
-		status.updateNodeStatus(nodeName, ready, reason, message)
-		return
+		err := status.updateNodeStatus(nodeName, ready, reason, message)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		return status.assertNodeStatus(nodeName, ready)
 	}
-	// When the network status looks ready, we still need to wait for a graceful time
+	// When the network status looks ready, we should check whether need to wait for a graceful time
 	now := time.Now()
 	startedTime := now.Sub(*startedAt)
-	go func() {
-		sleepTime := operatortypes.TimeBeforeRecoverNetwork - startedTime
-		time.Sleep(sleepTime)
-		if (sharedInfo.LastNetworkAvailable[nodeName]) != *startedAt {
-			log.V(1).Info(fmt.Sprintf("Last network unavailable time of node %s changed, expecting %v, got %v, goroutine exiting",
-				nodeName, *startedAt, sharedInfo.LastNetworkAvailable[nodeName]))
-			return
+	if startedTime >= operatortypes.TimeBeforeRecoverNetwork {
+		err := status.updateNodeStatus(nodeName, ready, reason, message)
+		if err != nil {
+			return reconcile.Result{}, err
 		}
-		log.Info(fmt.Sprintf("Setting status NetworkUnavailable to %v for node %s after %v", !ready, nodeName, sleepTime))
-		status.updateNodeStatus(nodeName, ready, reason, message)
-		return
-	}()
+		return status.assertNodeStatus(nodeName, ready)
+	}
+	if err := status.updateNodeStatus(nodeName, false, reason, message); err != nil {
+		return reconcile.Result{}, err
+	}
+	sleepTime := operatortypes.TimeBeforeRecoverNetwork - startedTime
+	log.Info(fmt.Sprintf("Waiting %v to double check network status for node %s", sleepTime, nodeName))
```

Comment:
How about using debug level for this log

Reply:
Done